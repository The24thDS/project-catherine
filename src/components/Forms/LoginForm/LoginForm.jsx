import React from "react";
import { withRouter } from "react-router-dom";
import {
  EuiForm,
  EuiFormRow,
  EuiFieldText,
  EuiButton,
  EuiLink,
  EuiText,
  EuiCheckbox,
  EuiFlexGroup
} from "@elastic/eui";

import Modal from "../../Modal";
import ForgotPasswordForm from "../ForgotPasswordForm";

import "../forms.sass";

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      emailInvalid: false,
      password: "",
      passwordInvalid: false,
      loading: false,
      formErrors: [],
      formMessage: "",
      rememberMe: false,
      forgotPasswordModal: false
    };
    this.serverUrl = process.env.REACT_APP_SERVER_URL;
  }

  onInputChange = event => {
    this.setState({ [event.target.name]: event.target.value, formErrors: "" });
  };

  validateUser = () => {
    if (this.state.user.length === 0) {
      this.setState({
        emailInvalid: true
      });
    } else {
      this.setState({
        emailInvalid: false
      });
    }
  };

  validatePassword = () => {
    if (this.state.password.length === 0) {
      this.setState({
        passwordInvalid: true
      });
    } else {
      this.setState({
        passwordInvalid: false
      });
    }
  };

  toggleForgotPasswordModal = () => {
    this.setState(prevState => ({
      ...prevState,
      forgotPasswordModal: !prevState.forgotPasswordModal
    }));
  };

  login = async () => {
    this.setState({ loading: true });
    const response = await fetch(`${this.serverUrl}/auth/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        user: this.state.user,
        password: this.state.password
      })
    });

    const data = await response.json();

    if (data.success === true) {
      if (this.state.rememberMe) {
        window.localStorage.setItem("token", data.message);
      } else {
        window.sessionStorage.setItem("token", data.message);
      }
      this.props.setLoggedIn(true, data.message);
      this.props.history.push("/events");
    } else {
      this.setState({
        loading: false,
        formErrors: data.reason || data.message
      });
    }
  };

  submitForm = async ev => {
    ev.preventDefault();
    const { emailInvalid, passwordInvalid, user, password } = this.state;
    if (!emailInvalid && !passwordInvalid && user.length && password.length) {
      await this.login();
    } else {
      this.setState({
        formErrors: "At least one of the inputs is invalid"
      });
    }
  };

  render() {
    const {
      email,
      emailInvalid,
      password,
      passwordInvalid,
      loading,
      formErrors,
      formMessage,
      rememberMe,
      forgotPasswordModal
    } = this.state;

    return (
      <>
        <EuiForm
          className="landing-form"
          isInvalid={formErrors.length > 0}
          error={formErrors}
          onSubmit={this.submitForm}
          component="form"
        >
          <EuiFormRow
            label="Email"
            isInvalid={emailInvalid}
            error="Can't be empty"
            onBlur={this.validateUser}
            fullWidth
          >
            <EuiFieldText
              name="email"
              type="email"
              value={email}
              onChange={this.onInputChange}
              fullWidth
            />
          </EuiFormRow>
          <EuiFormRow
            label="Password"
            isInvalid={passwordInvalid}
            error="Can't be empty"
            onBlur={this.validatePassword}
            fullWidth
          >
            <EuiFieldText
              name="password"
              type="password"
              value={password}
              onChange={this.onInputChange}
              fullWidth
            />
          </EuiFormRow>
          <EuiFormRow>
            <EuiCheckbox
              id="rememberMe"
              name="rememberMe"
              onChange={ev =>
                this.onInputChange({
                  target: { name: ev.target.name, value: !rememberMe }
                })
              }
              checked={rememberMe}
              label="Remember me"
            />
          </EuiFormRow>
          <EuiFormRow className="form-button" fullWidth>
            <EuiButton fill isLoading={loading} type="submit" fullWidth>
              Login
            </EuiButton>
          </EuiFormRow>
          <EuiFormRow className="form-button" fullWidth>
            <EuiButton
              onClick={this.toggleForgotPasswordModal}
              type="button"
              fullWidth
            >
              Forgot password?
            </EuiButton>
          </EuiFormRow>
          <EuiText className="form-message">{formMessage}</EuiText>
        </EuiForm>
        {forgotPasswordModal && (
          <Modal>
            <ForgotPasswordForm closeModal={this.toggleForgotPasswordModal} />
          </Modal>
        )}
      </>
    );
  }
}

export default withRouter(LoginForm);
