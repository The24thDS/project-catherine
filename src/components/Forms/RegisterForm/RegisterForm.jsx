import React from "react";
import {
  EuiForm,
  EuiFormRow,
  EuiFieldText,
  EuiButton,
  EuiText,
  EuiToolTip,
  EuiProgress,
  EuiFlexGroup,
  EuiFlexItem,
  EuiDatePicker,
} from "@elastic/eui";
import moment from "moment";
import * as yup from "yup";

import "../forms.sass";

import {
  validatePassword,
  calculatePasswordStrength,
} from "../../../utils/password";
import rightArrow from "../../../assets/arrow-right.svg";
import ServerRequest from "../../../utils/ServerRequest";

class RegisterForm extends React.Component {
  constructor() {
    super();
    this.state = {
      formInputs: {
        firstName: "",
        lastName: "",
        email: "",
        birthDate: moment().subtract(14, "years"),
        password: "",
      },
      formValidation: {
        firstName: true,
        lastName: true,
        email: true,
        birthDate: true,
        password: true,
      },
      passwordStrength: {
        value: 0,
        checking: false,
      },
      loading: false,
      formErrors: [],
      success: false,
    };
    this.serverUrl = process.env.REACT_APP_SERVER_URL;
  }

  onInputChange = ({ target }) => {
    this.setState((prevState) => ({
      formInputs: {
        ...prevState.formInputs,
        [target.name]: target.value,
      },
      formValidation: {
        firstName: true,
        lastName: true,
        email: true,
        birthDate: true,
        password: true,
      },
      formErrors: [],
    }));
  };

  validateForm = async () => {
    const registerFormSchema = yup.object({
      firstName: yup.string().trim().required("First Name is a required field"),
      lastName: yup.string().trim().required("Last Name is a required field"),
      email: yup.string().trim().email().required(),
      birthDate: yup
        .date()
        .max(
          moment().subtract(14, "years"),
          "You must be at least 14 years old."
        )
        .required("Date of Birth is a required field"),
      password: yup
        .string()
        .trim()
        .min(10)
        .notOneOf(
          this.state.formInputs.email !== ""
            ? [this.state.formInputs.email]
            : []
        )
        .test(
          "is-not-pwned",
          "Your password was exposed in a data breach. Stop using it! (source: haveibeenpwned.com)",
          async (value) => {
            this.setState((prevState) => ({
              passwordStrength: {
                ...prevState.passwordStrength,
                checking: true,
              },
            }));
            const result = await validatePassword(value);
            this.setState((prevState) => ({
              passwordStrength: {
                ...prevState.passwordStrength,
                checking: false,
              },
            }));
            return result;
          }
        )
        .required(),
    });
    try {
      const castedData = await registerFormSchema.validate(
        this.state.formInputs,
        {
          abortEarly: false,
        }
      );
      this.setState((prevState) => ({
        formInputs: {
          ...castedData,
          birthDate: prevState.formInputs.birthDate,
        },
      }));
      return true;
    } catch (error) {
      console.log(error);
      const formValidation = {};
      error.inner.forEach((errObj) => {
        formValidation[errObj.path] = false;
      });
      if (error.name === "ValidationError") {
        this.setState((prevState) => ({
          formErrors: error.errors.map(
            (error) => error.charAt(0).toUpperCase() + error.slice(1)
          ),
          formValidation: {
            ...prevState.formValidation,
            ...formValidation,
          },
        }));
      }
      return false;
    }
  };

  setPasswordStrength = (password) => {
    this.setState((prevState) => ({
      passwordStrength: {
        ...prevState.passwordStrength,
        value: calculatePasswordStrength(password),
      },
    }));
  };

  register = async () => {
    this.setState({
      loading: true,
    });
    const req = new ServerRequest(
      "/auth/register",
      "POST",
      { "Content-Type": "application/json" },
      this.state.formInputs
    );
    const response = await req.send();

    if (response.status === 201) {
      this.setState({
        loading: false,
        success: true,
      });
    } else if (response.status === 400) {
      this.setState((prevState) => ({
        loading: false,
        formErrors: "An account with this email already exists",
        formValidation: {
          ...prevState.formValidation,
          email: false,
        },
      }));
    }
  };

  submitForm = async (event) => {
    event.preventDefault();

    const valid = await this.validateForm();
    if (valid) {
      await this.register();
    } else {
      return false;
    }
  };

  render() {
    const {
      formInputs: { firstName, lastName, email, birthDate, password },
      formValidation,
      passwordStrength,
      loading,
      formErrors,
      success,
    } = this.state;

    const passwordHelpText = (
      <div>
        <EuiFlexGroup
          gutterSize="none"
          alignItems="center"
          justifyContent="spaceBetween"
          responsive={false}
          style={{ height: "20px" }}
        >
          <EuiFlexItem style={{ height: "100%" }} grow={1}>
            Strength:
          </EuiFlexItem>
          <EuiFlexItem grow={3}>
            <EuiProgress
              value={passwordStrength.value}
              max={100}
              aria-label="password strength"
              color={
                passwordStrength.value < 50
                  ? "danger"
                  : passwordStrength.value < 75
                  ? "accent"
                  : "secondary"
              }
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      </div>
    );

    const passwordTooltipContent = (
      <EuiText size="xs">
        <ul>
          <li>Must be at least 10 characters</li>
          <li>Must not be your email address</li>
          <li>Must not be a password exposed in a data breach</li>
        </ul>
      </EuiText>
    );

    return success ? (
      <EuiText
        style={{
          textAlign: "center",
          height: "calc(100% - 100px)",
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          alignItems: "center",
          padding: "40px",
          fontSize: "20px",
        }}
      >
        <h2>Account created</h2>
        <p>Check your inbox or spam folder for the activation email.</p>
      </EuiText>
    ) : (
      <EuiForm
        className="landing-form"
        isInvalid={formErrors.length > 0}
        error={formErrors}
        component="form"
        onSubmit={this.submitForm}
      >
        <EuiFlexGroup className="name-inputs" gutterSize="none">
          <EuiFlexItem>
            <EuiFormRow
              isInvalid={!formValidation.firstName}
              label="First Name"
              fullWidth
            >
              <EuiFieldText
                name="firstName"
                value={firstName}
                onChange={this.onInputChange}
                fullWidth
              />
            </EuiFormRow>
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiFormRow
              isInvalid={!formValidation.lastName}
              label="Last Name"
              fullWidth
            >
              <EuiFieldText
                name="lastName"
                value={lastName}
                onChange={this.onInputChange}
                fullWidth
              />
            </EuiFormRow>
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiFormRow label="Email" isInvalid={!formValidation.email} fullWidth>
          <EuiFieldText
            name="email"
            type="email"
            value={email}
            onChange={this.onInputChange}
            fullWidth
          />
        </EuiFormRow>
        <EuiFormRow
          fullWidth
          isInvalid={!formValidation.birthDate}
          label="Date of Birth"
        >
          <EuiDatePicker
            name="birthDate"
            dateFormat="YYYY-MM-DD"
            selected={birthDate}
            maxDate={moment().subtract(14, "years")}
            onChange={(ev) =>
              this.onInputChange({
                target: { name: "birthDate", value: ev },
              })
            }
            fullWidth
          />
        </EuiFormRow>
        <EuiFormRow
          label="Password"
          helpText={passwordHelpText}
          isInvalid={!formValidation.password}
          fullWidth
        >
          <EuiToolTip
            position="top"
            title="Stay safe â€” don't reuse passwords."
            content={passwordTooltipContent}
            anchorClassName="full-width"
          >
            <EuiFieldText
              name="password"
              type="password"
              value={password}
              minLength="10"
              onChange={(ev) => {
                this.onInputChange(ev);
                this.setPasswordStrength(ev.target.value);
              }}
              isLoading={passwordStrength.checking}
              fullWidth
            />
          </EuiToolTip>
        </EuiFormRow>
        <EuiFormRow className="form-button" fullWidth>
          <EuiButton fill isLoading={loading} type="submit" fullWidth>
            <img src={rightArrow} alt="next" />
          </EuiButton>
        </EuiFormRow>
      </EuiForm>
    );
  }
}

export default RegisterForm;
