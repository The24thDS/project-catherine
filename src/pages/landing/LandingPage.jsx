import React from "react";
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiTitle,
  EuiIcon
} from "@elastic/eui";

import "./landingPage.sass";
import blackLogo from "../../assets/logo-black.svg";

export default function LandingPage() {
  document.title = "Project Catherine | Landing";
  return (
    <EuiFlexGroup
      className="landing"
      alignItems="center"
      gutterSize="none"
      responsive={true}
      wrap={true}
    >
      <EuiFlexItem className="landing--information" component="section">
        <EuiText>
          <EuiTitle textTransform="uppercase">
            <h1>Project Catherine</h1>
          </EuiTitle>
          <EuiIcon className="logo" size="original" type={blackLogo} />
          <article>
            <p>
              But I must explain to you how all this mistaken idea of denouncing
              pleasure and praising pain was born and I will give you a complete
              account of the system, and expound the actual teachings of the
              great explorer of the truth, the master-builder of human
              happiness. No one rejects, dislikes, or avoids pleasure itself,
              because it is pleasure, but because those who do not know how to
              pursue pleasure rationally encounter consequences that are
              extremely painful. Nor again is there anyone who loves or pursues
              or desires to obtain pain of itself, because it is pain, but
              because occasionally circumstances occur in which toil and pain
              can procure him some great pleasure. To take a trivial example,
              which of us ever undertakes laborious physical exercise, except to
              obtain some advantage from it? But who has any right to find fault
              with a man who chooses to enjoy a pleasure that has no annoying
              consequences, or one who avoids a pain that
            </p>
          </article>
        </EuiText>
      </EuiFlexItem>
      <EuiFlexItem className="landing--forms" component="section">
        login/signup
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
