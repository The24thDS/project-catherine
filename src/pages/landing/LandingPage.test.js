import React from "react";
import { shallow } from "enzyme";

import LandingPage from "./LandingPage";
import RegisterForm from "../../components/Forms/RegisterForm";
import LoginForm from "../../components/Forms/LoginForm";

describe("Landing page", () => {
  let wrapper = shallow(<LandingPage location={{ state: undefined }} />);

  it("renders the details section", () => {
    expect(wrapper.find(".landing--information").length).toEqual(1);
    const title = wrapper.find("h1");
    expect(title.length).toEqual(1);
    expect(title.text()).toEqual("Project Catherine");
    expect(wrapper.find(".logo").length).toEqual(1);
  });

  it("renders the signup form", () => {
    expect(wrapper.find(RegisterForm).length).toEqual(1);
  });

  it("doesnt render the login form", () => {
    expect(wrapper.find(LoginForm).length).toEqual(0);
  });

  it("should change to the login form when you click on the login button", () => {
    const loginButton = wrapper.find(".forms-nav .forms-nav-button:last-child");
    loginButton.simulate("click");
    expect(wrapper.find(LoginForm).length).toEqual(1);
    expect(wrapper.find(RegisterForm).length).toEqual(0);
  });

  it("renders the login form when you are redirected from the activation page", () => {
    wrapper = shallow(
      <LandingPage location={{ state: { from: "activation" } }} />
    );
    expect(wrapper.find(LoginForm).length).toEqual(1);
    expect(wrapper.find(RegisterForm).length).toEqual(0);
  });
});
