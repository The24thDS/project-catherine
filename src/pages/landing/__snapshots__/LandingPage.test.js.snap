// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page snapshot renders 1`] = `
<div
  className="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap landing"
>
  <section
    className="euiFlexItem landing--information"
  >
    <h1
      className="euiTitle euiTitle--medium euiTitle--uppercase"
    >
      Project Catherine
    </h1>
    <img
      className="euiIcon euiIcon-isLoaded logo"
      src="logo-black.svg"
    />
    <article>
      <p>
        But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that
      </p>
    </article>
  </section>
  <section
    className="euiFlexItem landing--forms"
  >
    <nav
      className="forms-nav"
    >
      <button
        className="forms-nav-button  forms-nav-button--active"
        onClick={[Function]}
      >
        SIGNUP
      </button>
      <button
        className="forms-nav-button "
        onClick={[Function]}
      >
        LOGIN
      </button>
    </nav>
    <form
      className="euiForm landing-form"
      onSubmit={[Function]}
    >
      <div
        className="euiFlexGroup euiFlexGroup--directionRow euiFlexGroup--responsive name-inputs"
      >
        <div
          className="euiFlexItem"
        >
          <div
            className="euiFormRow euiFormRow--fullWidth"
            id="su1gy2hs-row"
          >
            <div
              className="euiFormRow__labelWrapper"
            >
              <label
                aria-invalid={false}
                className="euiFormLabel euiFormRow__label"
                htmlFor="su1gy2hs"
              >
                First Name
              </label>
            </div>
            <div
              className="euiFormRow__fieldWrapper"
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--fullWidth"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    className="euiFieldText euiFieldText--fullWidth"
                    id="su1gy2hs"
                    name="firstName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="euiFlexItem"
        >
          <div
            className="euiFormRow euiFormRow--fullWidth"
            id="u5gjc0j5-row"
          >
            <div
              className="euiFormRow__labelWrapper"
            >
              <label
                aria-invalid={false}
                className="euiFormLabel euiFormRow__label"
                htmlFor="u5gjc0j5"
              >
                Last Name
              </label>
            </div>
            <div
              className="euiFormRow__fieldWrapper"
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--fullWidth"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    className="euiFieldText euiFieldText--fullWidth"
                    id="u5gjc0j5"
                    name="lastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormRow euiFormRow--fullWidth"
        id="9zkg912u-row"
      >
        <div
          className="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid={false}
            className="euiFormLabel euiFormRow__label"
            htmlFor="9zkg912u"
          >
            Email
          </label>
        </div>
        <div
          className="euiFormRow__fieldWrapper"
        >
          <div
            className="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              className="euiFormControlLayout__childrenWrapper"
            >
              <input
                className="euiFieldText euiFieldText--fullWidth"
                id="9zkg912u"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormRow euiFormRow--fullWidth"
        id="t2zen7xj-row"
      >
        <div
          className="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid={false}
            className="euiFormLabel euiFormRow__label"
            htmlFor="t2zen7xj"
          >
            Date of Birth
          </label>
        </div>
        <div
          className="euiFormRow__fieldWrapper"
        >
          <span>
            <span
              className="euiDatePicker euiDatePicker--shadow"
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--fullWidth"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <div
                    className="react-datepicker-wrapper"
                  >
                    <div
                      className="react-datepicker__input-container"
                    >
                      <input
                        aria-label="2006-04-06"
                        className="euiDatePicker euiFieldText euiFieldText--fullWidth euiFieldText--withIcon"
                        disabled={false}
                        id="t2zen7xj"
                        name="birthDate"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        readOnly={false}
                        type="text"
                        value="2006-04-06"
                      />
                    </div>
                  </div>
                  <div
                    className="euiFormControlLayoutIcons"
                  >
                    <span
                      className="euiFormControlLayoutCustomIcon"
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </span>
          </span>
        </div>
      </div>
      <div
        className="euiFormRow euiFormRow--fullWidth"
        id="wyicc5u5-row"
      >
        <div
          className="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid={false}
            className="euiFormLabel euiFormRow__label"
            htmlFor="wyicc5u5"
          >
            Password
          </label>
        </div>
        <div
          className="euiFormRow__fieldWrapper"
        >
          <span
            className="euiToolTipAnchor full-width"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                className="euiFormControlLayout__childrenWrapper"
              >
                <input
                  className="euiFieldText euiFieldText--fullWidth"
                  minLength="10"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </span>
          <div
            className="euiFormHelpText euiFormRow__text"
            id="wyicc5u5-help"
          >
            <div>
              <div
                className="euiFlexGroup euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                style={
                  Object {
                    "height": "20px",
                  }
                }
              >
                <div
                  className="euiFlexItem euiFlexItem--flexGrow1"
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                >
                  Strength:
                </div>
                <div
                  className="euiFlexItem euiFlexItem--flexGrow3"
                >
                  <progress
                    aria-label="password strength"
                    className="euiProgress euiProgress--native euiProgress--m euiProgress--danger"
                    max={100}
                    value={0}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="euiFormRow euiFormRow--fullWidth form-button"
        id="6y0j6o8h-row"
      >
        <div
          className="euiFormRow__fieldWrapper"
        >
          <button
            className="euiButton euiButton--primary euiButton--fill euiButton--fullWidth"
            id="6y0j6o8h"
            onBlur={[Function]}
            onFocus={[Function]}
            type="submit"
          >
            <span
              className="euiButton__content"
            >
              <span
                className="euiButton__text"
              >
                <img
                  alt="next"
                  src="arrow-right.svg"
                />
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </section>
</div>
`;
